src/Services/Endpoint.php

```diff
@@ -42,19 +44,31 @@ class Endpoint extends SolariumEndpoint {
    * @param array $options
    *   Array of options for the endpoint. Currently,
    *   they are used by other functions of the endpoint.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
   public function __construct(array $options = []) {
-    // We intentionally want to override this options in case they are set in the parameter.
-    $options = array_merge($options, [
-      'scheme' => getenv('PANTHEON_INDEX_SCHEME'),
-      'host' => getenv('PANTHEON_INDEX_HOST'),
-      'port' => getenv('PANTHEON_INDEX_PORT'),
-      'path' => getenv('PANTHEON_INDEX_PATH'),
-      'core' => getenv('PANTHEON_INDEX_CORE'),
-      'schema' => getenv('PANTHEON_INDEX_SCHEMA'),
-      'collection' => NULL,
-      'leader' => FALSE,
-    ]);
+    $storage = \Drupal::entityTypeManager()->getStorage('search_api_server');
```

Comment:
It would be better to receive entity_type.manager service as argument in this __construct.

Reply:
@kporras07 agree, added